File: app/src/main/java/com/test/xyz/demo/presentation/projectlist/di/ProjectListFragmentComponent.java
package com.test.xyz.demo.presentation.projectlist.di;

import com.test.xyz.demo.presentation.projectlist.ProjectListFragment;
import com.test.xyz.demo.presentation.common.di.scope.ActivityScope;

import dagger.Subcomponent;

@ActivityScope
@Subcomponent(
        modules = {ProjectListFragmentModule.class}
)
public interface ProjectListFragmentComponent {
    void inject(ProjectListFragment projectListFragment);
}

File: app/src/main/kotlin/com/test/xyz/demo/presentation/projectlist/di/ProjectListFragmentComponent.kt
package com.test.xyz.demo.presentation.projectlist.di

import com.test.xyz.demo.presentation.common.di.scope.ActivityScope
import com.test.xyz.demo.presentation.projectlist.ProjectListFragment
import dagger.Subcomponent

@ActivityScope
@Subcomponent(modules = arrayOf(ProjectListFragmentModule::class))
interface ProjectListFragmentComponent {
    fun inject(projectListFragment: ProjectListFragment)
}

